go/runtime/client/watcher.go

```diff
@@ -149,14 +155,23 @@ func (w *blockWatcher) watch() {
 
 		// Find a new committee leader.
 		if current.Header.HeaderType == block.EpochTransition || !gotFirstBlock {
-			if err := w.committeeWatcher.EpochTransition(w.common.ctx, height); err != nil {
+			currentEpochNumber, err = w.common.consensus.EpochTime().GetEpoch(w.common.ctx, height)
```

Comment:
A minor nit - in case of an error (should never happen, but hey) this will mutate the `currentEpochNumber` with a potentially invalid epoch. It should instead not change the `currentEpochNumber` when an error occurs.